src/item_group.cpp

```diff
@@ -181,6 +181,8 @@ item Single_item_creator::create_single( const time_point &birthday, RecursionLi
         int qty = tmp.charges;
         if( modifier ) {
             qty = rng( modifier->charges.first, modifier->charges.second );
+        } else if( tmp.made_of_from_type( phase_id::LIQUID ) ) {
```

Comment:
What happens with solid items such as protein powder, etc?

Reply:
I'm not aware of any powders that come in containers larger than one unit; it's all small plastic bottles (if contained at all) I think?

Another approach I considered was making it so that anything in rigid containers would be maximised, but that'd mess with detergent powders and so on; this approach I _think_ covers all the cases for the fix.

Reply:
I'm pretty sure there are powders that spawn larger than one unit in containers. Even if there's currently not such an item, anyone could add one in the future.

If we do not ensure consistency for all items, it can surprise people when they add more spawns only to find out solid items behaved differently from liquid items.

Reply:
The issue is that powders (and similar) are already wildly inconsistent, which is why this is restricted to liquids.

Powders come in a wide array of containers, not all of which are rigid, and fill their containers to very different degrees. Liquids, by contrast, are simple: They always come in rigid, watertight containers, so pouring an infinite amount into the container until full is a ready fix for the specific problem of only one inventory unit (250 ml) of liquid spawning.

Reply:
> powders (and similar) are already wildly inconsistent

If I understand the code correctly, that inconsistency is caused by the default stack size of powders, not how containers are filled here. So either we change all items to spawn completely filling the container, or we keep the behavior to spawn one default stack in the container.

Reply:
Not all powders or powder-like substances have a full stack add up to 250 ml (c.f. maple syrup candies, for a specific example).

And again, they don't all use rigid containers; plastic bags, for example, support a full 6000 ml, which is generally way bigger than the contents are meant to be.

For this particular fix it makes sense to focus on liquids, because liquids are obviously and profoundly broken, while powders don't appear to be.

Reply:
If we only focus on fixing the immediate issue without ensuring consistency in the codebase, that will only make maintenance more and more difficult in the future. As you said plastic bags have a 6000ml full volume, so completely filling containers here is not intended. I suggest finding out whatever liquid that is not spawning correctly in its default container, and change its default stack size to match the container volume.

Reply:
If we only focus on fixing the immediate issue without ensuring consistency in the codebase, that will only make maintenance more and more difficult in the future. As you said plastic bags have a 6000ml full volume, so completely filling containers here is not intended. I suggest finding out whatever liquid that is not spawning correctly in its default container, and change its default stack size to match the container volume.

Reply:
That is not a workable solution, as some liquids come in more than one size; energy cola, for example, come in 1-unit cans and 8-unit large plastic bottles.

Reply:
The way crafting recipes works also means that the range of what stack sizes for liquids can be (c.f. martini).

Reply:
Well, I don't know, maybe make it fill all containers by default, and use a script to add charges to solid item spawn entries?